// grunt-makeHTML/tasks/libs/index.js
'use strict';

var glob = require('glob'),
    fs = require('graceful-fs'),
    path = require('path'),
	stripBom = require('strip-bom'),
	utf8 = 'utf-8';

/* move these to a utility library soon */
function readFile(src) {

    return stripBom(fs.readFileSync(src, utf8));

}

function writeFile(target, content) {

    return fs.writeFileSync(target, content, utf8);

}

function getViews(views) {

    var i = 0,
        html = "";

    for (i = 0; i < views.length; i++) {

        if (views[i] !== "") {

            html += readFile(views[i]) + "\r";

        }

    }

    return html;
}

function getScripts(scripts) {

    var i = 0,
        html = "";

    for (i = 0; i < scripts.length; i++) {

        if (scripts[i] !== "") {

            html += '<script src="' +
                    scripts[i] + '" id="' +
                    path.basename(scripts[i], path.extname(scripts[i]))
                    + '-script"></script>\r';

        }

    }

    return html;

}

function getStyles(styles) {

    var i = 0,
        html = "";

    for (i = 0; i < styles.length; i++) {

        if (styles[i] !== "") {

            html += '<link href="' +
                styles[i] + '" id="' +
                path.basename(styles[i], path.extname(styles[i]))
                + '-css" rel="stylesheet" />\r';

        }

    }

    return html;

}


exports.generate = function (opts) {

    var i = 0,
        html = "", output, indexSrc,
        styles = [];

    output = readFile(opts.indexSrc);
    
    output = output.replace(/<#spaCSS#>/g, getStyles(opts.styles));
    output = output.replace(/<#spaHTML#>/g, getViews(opts.srcFiles));
    output = output.replace(/<#spaScripts#>/g, getScripts(opts.scripts));

    output = output.replace(/<#spaCriticalCSS#>/g, readFile(opts.criticalCSS));
    
    writeFile(opts.dest, output);

};
