// grunt-makeHTML/tasks/makeHTML.js
'use strict';


var glob = require("glob"),
    fs = require("fs"),
    path = require("path");

var makeHTML = require("./libs/index.js");

module.exports = function (grunt) {

    grunt.registerMultiTask('makeHTML', 'Generates Compiled SPA HTML', function () {

        var options = this.options({
                            srcFiles : []
                        }),
            files  = [],
            views  = [],
            scripts = [],
            styles = [];

        options.src.forEach(function (element, index, array) {

            files = grunt.file.expand(path.resolve(element));
            
            for (var i = 0; i < files.length; i++) {

                views.push(files[i]);

            }

        });

        options.styles.forEach(function (element, index, array) {

            files = grunt.file.expand(element);

            for (var i = 0; i < files.length; i++) {

                styles.push(files[i]);
            }

        });

        options.scripts.forEach(function (element, index, array) {

            files = grunt.file.expand(element);

            for (var i = 0; i < files.length; i++) {

                scripts.push(files[i]);
            }


        });

        grunt.log.ok('writing compiled SPA HTML');

        makeHTML.generate({
            indexSrc: options.indexSrc,
            dest: options.dest,
            styles: styles,
            criticalCSS: criticalCSS,
            scripts: scripts,
            srcFiles: views
        });

        grunt.log.ok('File ' + options.dest + ' created');

    });

}
